docs/core/install/linux-ubuntu.md

```diff
@@ -0,0 +1,364 @@
+sudo dpkg -i packages-microsoft-prod.deb
+```
+
+[!INCLUDE [linux-apt-install-31](includes/linux-apt-install-21.md)]
+
+## APT 17.04 (not supported)
+
+Before installing .NET, you'll need to:
+
+- Add the Microsoft package signing key to the list of trusted keys.
+- Add the repository to the package manager.
+- Install required dependencies.
+
+> [!TIP]
+> Registering the key only needs to be done once per machine.
+
+Open a terminal and run the following commands.
+
+```bash
+wget https://packages.microsoft.com/config/ubuntu/17.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+sudo dpkg -i packages-microsoft-prod.deb
+```
+
+[!INCLUDE [linux-apt-install-31](includes/linux-apt-install-21.md)]
+
+## APT 16.10 (not supported)
+
+Before installing .NET, you'll need to:
+
+- Add the Microsoft package signing key to the list of trusted keys.
+- Add the repository to the package manager.
+- Install required dependencies.
+
+> [!TIP]
+> Registering the key only needs to be done once per machine.
+
+Open a terminal and run the following commands.
+
+```bash
+wget https://packages.microsoft.com/config/ubuntu/16.10/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+sudo dpkg -i packages-microsoft-prod.deb
+```
+
+[!INCLUDE [linux-apt-install-31](includes/linux-apt-install-21.md)]
+
+## APT 16.04 (supported)
+
+Before installing .NET, you'll need to:
+
+- Add the Microsoft package signing key to the list of trusted keys.
+- Add the repository to the package manager.
+- Install required dependencies.
+
+> [!TIP]
+> Registering the key only needs to be done once per machine.
+
+Open a terminal and run the following commands.
+
+```bash
+wget https://packages.microsoft.com/config/ubuntu/16.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+sudo dpkg -i packages-microsoft-prod.deb
+```
+
+[!INCLUDE [linux-apt-install-31](includes/linux-apt-install-31.md)]
+
+## APT update the SDK or runtime
+
+When a new patch release is available for .NET Core, you can simply upgrade it through APT with the following commands:
+
+```bash
+sudo apt-get update
+sudo apt-get upgrade
+```
+
+## APT troubleshooting
+
+This section provides information on common errors you may get while using [APT to install .NET Core](#apt-advanced-package-tool).
+
+### Unable to locate
+
+If you receive an error message similar to **Unable to locate package {netcore-package}**, run the following commands.
+
+```bash
+sudo dpkg --purge packages-microsoft-prod && sudo dpkg -i packages-microsoft-prod.deb
+sudo apt-get update
+sudo apt-get install {netcore-package}
+```
+
+If that doesn't work, you can run a manual install with the following commands:
+
+```bash
+sudo apt-get install -y gpg
+wget -O - https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor -o microsoft.asc.gpg
+sudo mv microsoft.asc.gpg /etc/apt/trusted.gpg.d/
+wget https://packages.microsoft.com/config/ubuntu/{ubuntu-version}/prod.list
```

Comment:
I'm not too keen on `{netcore-package}` at the end, but it's probably fine since it's the last line--it's easy to see and leave out of my copy-paste. But `{ubuntu-version}` in the middle of a download line, early in the command list seems much worse.

Reply:
I added some intro text to it. Considering this is troubleshooting content, I think the user will be paying more attention to the wording and intro text in this section versus the install sections which are a bit more "blast away" friendly.

Reply:
That's a good point. I think though it would be a little bit nicer to use variable formatting, so the user can set up the vars and still simply copy-paste the script. (To keep it simple, I suppose there's no need to call it out, but I'd definitely take advantage of it if I needed to follow these instructions.)

> - `${netcore_package}`\
This represents the .NET Core package you're installing, such as `20.04`. This is in the `sudo apt-get install` command.

> ```
> ...
> wget https://packages.microsoft.com/config/ubuntu/${ubuntu_version}/prod.list
> ...
> sudo apt-get install ${netcore_package}
> ```

(or `$netcore_package`, not sure which type of delimiting makes it clearer.)